src/_P076_HLW8012.ino

```diff
@@ -67,27 +56,40 @@ struct p076_DevicePinSettings {
 
 };
 
-typedef struct p076_DevicePinSettings P076_DevicePinSettings;
+typedef struct p076_PredefinedDevice P076_PredefinedDevice;
 
+//When adding new device increase counter, too
 const int p076_PinSettingsCount = 10;
-P076_DevicePinSettings p076_PredefinedPinSettings[p076_PinSettingsCount] =
+const char string_Custom[]    PROGMEM = "Custom";
+const char string_Sonoff[]    PROGMEM = "Sonoff Pow";
+const char string_Huafan[]    PROGMEM = "Huafan SS";
+const char string_KMC[]       PROGMEM = "KMC 70011";
+const char string_Aplic[]     PROGMEM = "Aplic WDP303075";
+const char string_SK03[]      PROGMEM = "SK03 Outdoor";
+const char string_BlitzWolf[] PROGMEM = "BlitzWolf SHP";
+const char string_Teckin[]    PROGMEM = "Teckin";
+const char string_TeckinUS[]  PROGMEM = "Teckin US";
+const char string_Gosund[]    PROGMEM = "Gosund SP1 v23";
+static const P076_PredefinedDevice p076_PredefinedDevices[] PROGMEM =
 {
-   //Device_Name,SEL_PIN, CF_PIN, CF1_PIN, Current_Read, CF_Trigger,  CF1_Trigger
-   //HLW8012
-   { 0, "Custom",           0,      0,     0,           LOW,        LOW,    LOW},
-   { 1, "Sonoff Pow",       5,     14,     3,          HIGH,     CHANGE, CHANGE},
-   { 2, "Huafan SS",       13,     14,    12,          HIGH,     CHANGE, CHANGE},
-   { 3, "KMC 70011",       12,      4,     5,          HIGH,     CHANGE, CHANGE},
-   { 4, "Aplic WDP303075", 12,      4,     5,           LOW,     CHANGE, CHANGE},
-   { 5, "SK03 Outdoor",    12,      4,     5,           LOW,     CHANGE, CHANGE},
-
-   //BL093
-   { 6, "BlitzWolf SHP",   12,      5,    14,           LOW,    FALLING, CHANGE},
-   { 7, "Teckin",          12,      4,     5,           LOW,    FALLING, CHANGE},
-   { 8, "Teckin US",       12,      5,    14,           LOW,    FALLING, CHANGE},
-   { 9, "Gosund SP1 v23",  12,      4,     5,           LOW,    FALLING, CHANGE}
+   //Device_Name,SEL_PIN, CF_PIN, CF1_PIN, Current_Read, CF_Trigger, CF1_Trigger
+   //HLW8012 Devices
+   { 0, string_Custom,      0,      0,     0,       LOW,        LOW,    LOW},
+   { 1, string_Sonoff,      5,     14,     3,      HIGH,     CHANGE, CHANGE},
+   { 2, string_Huafan,     13,     14,    12,      HIGH,     CHANGE, CHANGE},
+   { 3, string_KMC,        12,      4,     5,      HIGH,     CHANGE, CHANGE},
+   { 4, string_Aplic,      12,      4,     5,       LOW,     CHANGE, CHANGE},
+   { 5, string_SK03,       12,      4,     5,       LOW,     CHANGE, CHANGE},
+
+   //BL093 Devices
+   { 6, string_BlitzWolf,  12,      5,    14,       LOW,    FALLING, CHANGE},
+   { 7, string_Teckin,     12,      4,     5,       LOW,    FALLING, CHANGE},
+   { 8, string_TeckinUS,   12,      5,    14,       LOW,    FALLING, CHANGE},
+   { 9, string_Gosund,     12,      4,     5,       LOW,    FALLING, CHANGE}
 };
 
+P076_PredefinedDevice p076_SelectedPredefinedDevice;
```

Comment:
Please either declare this on the stack, inside the function where you use it, or make it an allocation on the heap when needed (using a pointer and new... )
Like this, it is created and always consuming memory, even if the plugin is not active.